scripts/deploy_factory_dev.sh

```diff
@@ -0,0 +1,229 @@
+#!/bin/bash
+#### --------------------------------------------
+#### NOTE: The following flows are supported in this file, for testing!
+# - deploy factory v2
+# - create new DAO
+# - create 1-3 proposals
+# - upgrade to factory v3 
+# - submit DAO code v2 
+# - submit DAO metadata v2
+# - submit DAO code v3 
+# - submit DAO metadata v3
+# - upgrade DAO to v3 
+# - check if proposals return
+#### --------------------------------------------
+set -e
+
+if [ -d "res" ]; then
+  echo ""
+else
+  mkdir res
+fi
+
+cd "`dirname $0`"
+
+if [ -z "$KEEP_NAMES" ]; then
+  export RUSTFLAGS='-C link-arg=-s'
+else
+  export RUSTFLAGS=''
+fi
+
+# TODO: Change to the official approved commit:
+COMMIT_V3=596f27a649c5df3310e945a37a41a957492c0322
+# git checkout $COMMIT_V3
+
+# build the things
+cargo build --all --target wasm32-unknown-unknown --release
+cp ../target/wasm32-unknown-unknown/release/*.wasm ../res/
+
+export NEAR_ENV=testnet
+export FACTORY=testnet
+
+if [ -z ${NEAR_ACCT+x} ]; then
+  # export NEAR_ACCT=sputnikv2.$FACTORY
+  export NEAR_ACCT=sputnikpm.$FACTORY
+else
+  export NEAR_ACCT=$NEAR_ACCT
+fi
+
+export FACTORY_ACCOUNT_ID=factory2.$NEAR_ACCT
+# export DAO_ACCOUNT_ID=croncat.sputnikv2.$FACTORY
+export DAO_ACCOUNT_ID=sputnikdao-dev-1644971179.factory2.sputnikpm.testnet
+export MAX_GAS=300000000000000
+export GAS_100_TGAS=100000000000000
+export GAS_150_TGAS=150000000000000
+
+# #### --------------------------------------------
+# #### Account & Data management for setup
+# #### --------------------------------------------
+# near call $FACTORY_ACCOUNT_ID delete_contract '{"code_hash":"Cn8QJe3SJk36b9NZky4SQuKtu9L2ktYYTCBdy2qKJ8Bq"}' --accountId $FACTORY_ACCOUNT_ID --gas $GAS_100_TGAS
+# near call $FACTORY_ACCOUNT_ID delete_contract '{"code_hash":"8RMeZ5cXDap6TENxaJKtigRYf3n139iHmTRe8ZUNey6N"}' --accountId $FACTORY_ACCOUNT_ID --gas $GAS_100_TGAS
+# near call $FACTORY_ACCOUNT_ID delete_contract '{"code_hash":"ZGdM2TFdQpcXrxPxvq25514EViyi9xBSboetDiB3Uiq"}' --accountId $FACTORY_ACCOUNT_ID --gas $GAS_100_TGAS
+# near call $FACTORY_ACCOUNT_ID delete_contract '{"code_hash":"6SQymHtmezR3u9zAtLBQdb8RWCXxwxnigqSH2mRTxecB"}' --accountId $FACTORY_ACCOUNT_ID --gas $GAS_100_TGAS
```

Comment:
why do we have 5 examples of the same thing?

Reply:
These are actually 3 different on-chain versions deployed, my goal here is to catalog real contracts so I can triple check the process of upgrades for all scenarios. think of it as coverage for v1, v2, v2-recent & then local